docs/docs/multirole.md

```diff
@@ -0,0 +1,337 @@
+* All services are updated to use the group-role
+* `enforceRole` is enabled for all top-level groups
+* The `--new_group_enforce_role` command-line parameter is set to `"top"`.
+
+# Service role field
+
+Each Marathon service now contains a "role" field. By default, this "role" field defaults to the role specified by the command line parameter `--mesos_role`, which in turn is `*` when not specified, by default. For example, if the command line parameter `--mesos_role` is `"slave_public"` (as is the default in DC/OS), and the following service is posted to Marathon `/v2/apps`:
+
+```
+{
+  "id": "/dev/bigbusiness",
+  "cmd": "sleep 3600",
+  "instances": 10,
+  "cpus": 0.05,
+  "mem": 128
+}
+```
+
+... then, the resulting Marathon service will receive the default role `"slave_public",` as follows:
+
+```
+{
+  "id": "/dev/bigbusiness",
+  "cmd": "set; sleep 3600",
+  "instances": 10,
+  "cpus": 0.05,
+  "mem": 128,
+  "role": "slave_public"
+}
+```
+
+Marathon will then proceed to revive offers for the role `"slave_public",` and match offers allocated to the role `"slave_public"` in order to launch this app.
+
+# Group role
+
+The group role is the name of the top-level group, as illustrated by the following table:
+
+| Full service id | Marathon group | Group-role |
+| --------------- | -------------- | --------- |
+| /dev/bigbusiness | /dev | "dev" |
+| /bigbusiness | / | N/A |
+| /dev/team1/bigbusiness | /dev/team1 | "dev" |
+| /frontend/ui | /frontend | "frontend" |
+
+Note that services placed directly in the root group do not have a group-role. A service may only be assigned one of two possible roles: the default role (that which is specified by `--mesos_role`), or, the group role.
+
+# Migrating existing services
+
+When an existing service role is modified from using the default Mesos role to using the group role, this is referred to as "migrating the service to multi-role".
+
+Migrating ephemeral services is straightforward: update the role, and Marathon relaunches the service, accepting and using resources from offers for the new role. However, there is a caveat when migrating services with reservations.
+
+## Ephemeral service multi-role migration
+
+In order to migrate the bigbusiness app described earlier in this document from using the default Mesos role to the group role, we can simply post the following:
+
+```
+echo '{"role": "dev"} | curl -X PATCH http://local:8080/v2/apps/dev/bigbusiness -H "Content-Type: application/json" --data "@-"
+```
+
+This will trigger a deployment where all of the old bigbusiness instances running as the role `"slave_public"` are killed, and replaced by new instances running as the role "dev". Further, Marathon will update its FrameworkInfo information with Mesos to also include the role "dev", so that it can begin receiving offers allocated to the role. This may require the Mesos create role permission if the role does not already exist. See the section below on Mesos permission caveats.
```

Comment:
```suggestion
This will trigger a deployment where all of the old `bigbusiness` instances running as the role `"slave_public"` are killed, and replaced by new instances running as the role `"dev"`. Further, Marathon will update its `FrameworkInfo` information with Mesos to also include the role `"dev"`, so that it can begin receiving offers allocated to the role. This may require the Mesos `create role` permission if the role does not already exist. See the section below on Mesos permission caveats.
```